### DISCUSSION TITLE
Add colours to theme object in Typescript file

### DISCUSSION THREAD

**Discussion by I-am-tk** [mantinedev/mantine#1147] [Q&A] [answered]
This is how I want my color object to be. 

```
 theme = {
       colors: {
            warning: {
                  "25":  "",
                  "50" : ""
                 "100":  "",
                 "200" : ""
                 "300":  "",
                  "400" : ""
                  "500":  "",
                  "600" : "",
                  "700": "",
                  "800": "",
                  "900": ""
             }
       }
   }
```

And then provide the theme object to `MantineProvider`

```
<MantineProvider theme={theme}>
        .....
</MantineProvider>
```

Since `theme` object should be of `MantineThemeOverride` type. I even tried TypeScript module declaration. Still, I am getting some Typescript errors. I am new to Typesctipt. How to add type correctly?

I have one more doubt that for the default colors provided by Mantine like `dark`, type of which is `Tuple<string,10>`. And it is used by the Mantine when `colorScheme="dark"` is used. So will it break if I override the default `dark` color with a type other than `Tuple`, when  `colorScheme` is set  to `"dark"`.

**Comment by rtivital** [comment, marked as answer]
Color cannot be an object, it is required for them to be an array. Mantine component will not work with colors as object.

**Reply by I-am-tk** [reply]
Thanks, I can make it an array of length 12.

**Comment by I-am-tk** [comment]
Hey @rtivital, please see the code I don't know how to add the correct type. The theme object is giving an error. Here is the link to the code https://codesandbox.io/s/interesting-perlman-tlet7n?file=/src/App.tsx.

**Reply by rtivital** [reply]
https://codesandbox.io/s/dreamy-bose-z6smd9

**Comment by I-am-tk** [comment]
Thanks for the quick response but If I do like this I don't get IntelliSense. Is there any other way so that I can get typescript suggestions as well?

**Reply by rtivital** [reply]
I think we've already had this conversation yesterday

**Comment by I-am-tk** [comment]
I am so sorry, I am not asking questions properly. I have been asking this question again and again.

You had shared the link to add custom `color` properties. Where we had to define the type of `MantineThemeColorsOverride`
```
export interface MantineThemeColorsOverride {
    colors: Record<ExtendedCustomColors, Tuple<string, 10>>;
 }
```
The same for `other` property in the theme object. 

```
export interface MantineThemeColorsOverride {
    colors: Record<ExtendedCustomColors, Tuple<string, 10>>;
  }
```

Which type should be defined to add custom spacing and boxShadow?

**Reply by rtivital** [reply]
I do not know, you can try setting MantineSize type, but this may break in future
